### PR TITLE
`azurerm_managed_application` - add support for managed identity

### PR DESCRIPTION
**Pull request by hqhqhqhqhqhqhqhqhqhqhq**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

This PR adds support for managed identity to the `azurerm_managed_application` resource. The identity functionality was already implemented in the codebase but was missing from the resource schema and documentation. This change enables users to configure system-assigned, user-assigned, or both types of managed identities for their managed applications.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevant documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass.

image
https://hashicorp.teamcity.com/buildConfiguration/TF_AzureRM_AZURERM_SERVICE_PUBLIC_MANAGEDAPPLICATIONS/682763?buildTab=tests

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_managed_application` - support for the `identity` property

## Rollback Plan

If a change needs to be reverted, we will publish an updated version of the provider.

## Changes to Security Controls

This PR adds managed identity support to managed applications, which enhances security by enabling:
- System-assigned identities for automatic credential management
- User-assigned identities for shared identity scenarios
- Both identity types for comprehensive access control

The identity functionality follows the same patterns used by other Azure resources in the provider and maintains consistency with Azure's managed identity best practices.

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

### CODE REVIEW

File: internal/services/managedapplications/managed_application_resource_test.go [hunk 1]
```diff
@@ -153,6 +153,58 @@ func TestAccManagedApplication_tags(t *testing.T) {
...
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.systemAssignedIdentity(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
```
Comment (ms-henglu) [inline internal/services/managedapplications/managed_application_resource_test.go:174]: Can you add more steps to test it can change from system assigned identity to user assigned identity, then change back to none identity?

**Review by ms-henglu — CHANGES_REQUESTED** [review]
Thanks for the PR! LGTM, just one minor suggestion.

**Review by ms-henglu — APPROVED** [review]
Thanks for the PR! LGTM.

File: internal/services/managedapplications/managed_application_resource.go [hunk 2]
```diff
@@ -462,3 +487,67 @@ func compactParameterOrOutputValue(v interface{}) (string, error) {
...
+	out := &applications.Identity{
+		Type: &resourceType,
+	}
+
+	if expanded.Type == identity.TypeUserAssigned || expanded.Type == identity.TypeSystemAssignedUserAssigned {
+		userAssignedIdentities := make(map[string]applications.UserAssignedResourceIdentity)
+		for k, v := range expanded.IdentityIds {
+			userAssignedIdentities[k] = applications.UserAssignedResourceIdentity{
+				PrincipalId: v.PrincipalId,
+				TenantId:    v.ClientId,
```
Comment (liuwuliuyun) [inline internal/services/managedapplications/managed_application_resource.go]: Hi, just curious—why is it assigned as v.clientid instead of v.tenentid?

File: internal/services/managedapplications/managed_application_resource.go [hunk 3]
```diff
@@ -462,3 +487,67 @@ func compactParameterOrOutputValue(v interface{}) (string, error) {
...
+
+		if input.TenantId != nil {
+			config.TenantId = *input.TenantId
+		}
+
+		identityIds := make(map[string]identity.UserAssignedIdentityDetails)
+		if input.UserAssignedIdentities != nil {
+			for k, v := range *input.UserAssignedIdentities {
+				identityIds[k] = identity.UserAssignedIdentityDetails{
+					ClientId:    v.TenantId,
```
Comment (liuwuliuyun) [inline internal/services/managedapplications/managed_application_resource.go]: Same here

File: website/docs/r/managed_application.html.markdown [hunk 4]
```diff
@@ -103,6 +105,18 @@ The `plan` block exports the following:
...
 
+---
+
+An `identity` block supports the following:
+
+* `type` - (Required) Specifies the type of Managed Service Identity that should be configured on this Managed Application. Possible values are `SystemAssigned`, `UserAssigned`, and `SystemAssigned, UserAssigned` (to enable both).
+
+* `identity_ids` - (Optional) Specifies a list of User Assigned Managed Identity IDs to be assigned to this Managed Application.
+
+~> **Note:** When `type` is set to `SystemAssigned`, The assigned `principal_id` and `tenant_id` can be retrieved after the Managed Application has been created. More details are available below.
```
Comment (liuwuliuyun) [inline website/docs/r/managed_application.html.markdown]: It mentions that more details are available below, but I can't seem to find them in this PR.

**Review by liuwuliuyun — COMMENTED** [review]
Just a few tweaks or a quick explanation, and we're all set.

File: internal/services/managedapplications/managed_application_resource.go (hunk 2 above)
Comment (hqhqhqhqhqhqhqhqhqhqhq) [reply internal/services/managedapplications/managed_application_resource.go]: Good catch - this was a bug. The API's UserAssignedResourceIdentity only has PrincipalId and TenantId fields (no ClientId). I've fixed the mapping to only use PrincipalId since ClientId isn't exposed by the managed applications API.

File: website/docs/r/managed_application.html.markdown (hunk 4 above)
Comment (hqhqhqhqhqhqhqhqhqhqhq) [reply website/docs/r/managed_application.html.markdown]: Thx, updated

**Review by liuwuliuyun — APPROVED** [review]
LGTM!

File: internal/services/managedapplications/managed_application_resource.go [hunk 5]
```diff
@@ -462,3 +487,63 @@ func compactParameterOrOutputValue(v interface{}) (string, error) {
...
+
+	return out, nil
+}
+
+func flattenManagedApplicationIdentity(input *applications.Identity) ([]interface{}, error) {
+	var config *identity.SystemAndUserAssignedMap
+
+	if input != nil {
+		config = &identity.SystemAndUserAssignedMap{
+			Type:        identity.Type(*input.Type),
```
Comment (sreallymatt) [inline internal/services/managedapplications/managed_application_resource.go]: Given this is a pointer we should nil check `Type` or use `pointer.From` if it's acceptable for an empty string to be set here

File: website/docs/r/managed_application.html.markdown [hunk 6]
```diff
@@ -103,6 +105,16 @@ The `plan` block exports the following:
...
 
+---
+
+An `identity` block supports the following:
+
+* `type` - (Required) Specifies the type of Managed Service Identity that should be configured on this Managed Application. Possible values are `SystemAssigned`, `UserAssigned`, and `SystemAssigned, UserAssigned` (to enable both).
+
+* `identity_ids` - (Optional) Specifies a list of User Assigned Managed Identity IDs to be assigned to this Managed Application.
+
+~> **Note:** The `identity_ids` is required when `type` is set to `UserAssigned` or `SystemAssigned, UserAssigned`.
```
Comment (sreallymatt) [inline website/docs/r/managed_application.html.markdown]:
```suggestion
~> **Note:** The `identity_ids` argument is required when `type` is set to `UserAssigned` or `SystemAssigned, UserAssigned`.
```

**Review by sreallymatt — CHANGES_REQUESTED** [review]
Thanks @hqhqhqhqhqhqhqhqhqhqhq - just a couple comments, the added [test is also failing](https://hashicorp.teamcity.com/buildConfiguration/TF_AzureRM_AZURERM_SERVICE_PUBLIC_MANAGEDAPPLICATIONS/639810?buildTab=tests)

File: internal/services/managedapplications/managed_application_resource.go (hunk 5 above)
Comment (hqhqhqhqhqhqhqhqhqhqhq) [reply internal/services/managedapplications/managed_application_resource.go]: thanks, `Type` is now nil-checked

File: website/docs/r/managed_application.html.markdown (hunk 6 above)
Comment (hqhqhqhqhqhqhqhqhqhqhq) [reply website/docs/r/managed_application.html.markdown]: thx, updated

**Comment by hc-github-team-tf-azure** [comment]
**Testing in Beta version enabled**
  
Build: [701455](https://hashicorp.teamcity.com/viewLog.html?buildId=701455)
PR: #30725

**Total:** 9
**Passed:** 9
**Failed:** 0
**Test Duration:** 0h 21m 44s

<details>
<summary>Test Details</summary>

<table>
<tr><td><b>Status</b></td><td><b>Test Name</b></td><td><b>Duration</b></td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_basic</td><td>275.351s</td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_requiresImport</td><td>281.356s</td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_parameterValuesSecureString</td><td>289.852s</td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_allSupportedParameterValuesTypes</td><td>292.799s</td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_userAssignedIdentity</td><td>298.177s</td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_systemAssignedUserAssignedIdentity</td><td>303.385s</td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_parameterValues</td><td>389.275s</td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_tags</td><td>405.773s</td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_plan</td><td>492.989s</td></tr></table>
</details>

**Comment by hc-github-team-tf-azure** [comment]
Build: [701453](https://hashicorp.teamcity.com/viewLog.html?buildId=701453)
PR: #30725

**Total:** 10
**Passed:** 10
**Failed:** 0
**Test Duration:** 0h 13m 48s

<details>
<summary>Test Details</summary>

<table>
<tr><td><b>Status</b></td><td><b>Test Name</b></td><td><b>Duration</b></td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_requiresImport</td><td>287.701s</td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_parameterValuesSecureString</td><td>306.449s</td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_userAssignedIdentity</td><td>312.811s</td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_systemAssignedUserAssignedIdentity</td><td>317.499s</td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_allSupportedParameterValuesTypes</td><td>318.862s</td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_basic</td><td>324.711s</td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_parameterValues</td><td>415.467s</td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_tags</td><td>426.902s</td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_plan</td><td>597.936s</td></tr>
<tr><td>✅ PASS</td><td>TestAccManagedApplication_identity</td><td>753.915s</td></tr></table>
</details>

**Comment by hqhqhqhqhqhqhqhqhqhqhq** [comment]
Hi @sreallymatt, I've addressed the outstanding feedback, and the current checks and TeamCity run are passing. Could you please take another look when you have a chance? Thanks.

**Review by sreallymatt — APPROVED** [review]
Thanks @hqhqhqhqhqhqhqhqhqhqhq - LGTM ✅
